userspace/libscap/engine/gvisor/scap_gvisor.cpp

```diff
@@ -221,10 +234,67 @@ static void accept_thread(int listenfd, int epollfd)
 
 int32_t engine::start_capture()
 {
+	//
+	// Retrieve all running sandboxes
+	// We will need to recreate a session for each of them
+	//
+	runsc_result exisiting_sandboxes_res = runsc_list();
+	if(exisiting_sandboxes_res.error)
+	{
+		strlcpy(m_lasterr, "Error listing running sandboxes", SCAP_LASTERR_SIZE);
+		return SCAP_FAILURE;
+	}
+	std::vector<std::string> &existing_sandboxes = exisiting_sandboxes_res.output;
+
+	// Start accepting connections
 	m_accept_thread = std::thread(accept_thread, m_listenfd, m_epollfd);
 	m_accept_thread.detach();
 
 	m_capture_started = true;
+	
+	for(const auto& sandbox : existing_sandboxes)
+	{
+		// Since they were already running, we need to force the creation
+		runsc_result trace_create_res = runsc_trace_create(sandbox, true);
+		if(trace_create_res.error)
+		{
+			snprintf(m_lasterr, SCAP_LASTERR_SIZE, "Cannot create session for sandbox %s", sandbox.c_str());
+			return SCAP_FAILURE;
+		}
+	}
+
+
+	// Catch all sandboxes that might have been created in the meantime
+	runsc_result new_sandboxes_res = runsc_list();
```

Comment:
Is this sort of never-ending?
I mean, what if while we run the `new_sandboxes` check, new sandboxes are created?
If we have no way to guarantee that no sandboxes are lost, i think we can skip this check, WDYT?

Reply:
> What if while we run the new_sandboxes check, new sandboxes are created?

In that case they will connect to the socket because it is already been created. The problem is only if new sandboxes are created _before_ the socket is up.

Reply:
Exactly!